content/courses/tokens-and-nfts/nfts-with-metaplex.md

```diff
@@ -626,92 +716,96 @@ await airdropIfRequired(
-
-// Upload new offchain metadata
-const uploadMetadataOutput = await metaplex.nfts().uploadMetadata({
-  name: updatedNftData.name,
-  symbol: updatedNftData.symbol,
-  description: updatedNftData.description,
-  image: imageUri,
+const NFTImagePath = path.resolve(__dirname, "nft.png");
+
+const buffer = readFileSync(NFTImagePath);
+let file = createGenericFile(buffer, NFTImagePath, {
+  contentType: "image/png",
 });
 
-const updatedUri = uploadMetadataOutput.uri;
+// upload new image and get image uri
+const [image] = await umi.uploader.upload([file]);
+console.log("image uri:", image);
 
-// update the NFT metadata
-const { response } = await metaplex.nfts().update(
-  {
-    nftOrSft: nft,
-    uri: updatedUri,
+// upload updated offchain json using irys and get metadata uri
+const uri = await umi.uploader.uploadJson({
+  name: "Updated ",
+  symbol: "UPDATED",
+  description: "Updated Description",
+  image,
+});
+console.log("NFT offchain metadata URI:", uri);
+
+// Load the NFT using the mint address
+const mint = UMIPublicKey("Zxd9TmtBHQNti6tJxtx1AKYJFykNUwJL4rth441CjRd");
+const nft = await fetchMetadataFromSeeds(umi, { mint });
+
+await updateV1(umi, {
+  mint,
+  authority: umi.identity,
+  data: {
+    ...nft,
+    sellerFeeBasisPoints: 100, // 1%
+    name: "Updated Asset",
   },
-  { commitment: "finalized" },
-);
+  primarySaleHappened: true,
+  isMutable: true,
+}).sendAndConfirm(umi);
 
-console.log(
-  `NFT updated with new metadata URI: ${getExplorerLink(
-    "transaction",
-    response.signature,
-    "devnet",
-  )}`,
-);
+let explorerLink = getExplorerLink("address", mint, "devnet");
+console.log(`NFT updated with new metadata URI: ${explorerLink}`);
 
 console.log("✅ Finished successfully!");
 ```
 
 Run `npx esrun update-metaplex-nft.ts`. You should see something like:
 
-```typescript
+```bash
 % npx esrun update-metaplex-nft.ts
-Loaded user: 4DRi8LxWhAWf9LwUuyb3Rz4Y4USEKzjdvQBWTyrkJtwg
-image uri: https://arweave.net/nd6rxpoGfPmB2gTutEgnH5B76CFlDxgM6cZ7BcUmFis
-NFT updated with new metadata URI: https://explorer.solana.com/tx/4eiuMFLeku7ArV5xkRTPKxBXjfjQsSVGFZimMkMaMxK1d21qptQE5kgKiuob6J852m4GuoMqcjz1ee4oy7fXVEWU?cluster=devnet
+
+Loaded user: 4kg8oh3jdNtn7j2wcS7TrUua31AgbLzDVkBZgTAe44aF
+image uri: https://arweave.net/dboiAebucLGhprtknDQnp-yMj348cpJF4aQul406odg
+NFT offchain metadata URI: https://arweave.net/XEjo-44GHRFNOEtPUdDsQlW5z1Gtpk2Wv0HvR8ll1Bw
+NFT updated with new metadata URI: https://explorer.solana.com/address/Zxd9TmtBHQNti6tJxtx1AKYJFykNUwJL4rth441CjRd?cluster=devnet
 ✅ Finished successfully!
 ```
 
 Inspect the updated NFT on Solana Explorer! Just like previously, if you have
 any issues, you should fix them yourself, but if needed the
-[solution code](https://github.com/solana-developers/professional-education/blob/main/labs/update-metaplex-nft.ts)
+[solution code](https://github.com/solana-developers/professional-education/blob/main/labs/metaplex-umi/update-nft.ts)
 is available.
```

Comment:
Add an image here of what Explorer should look like. See CONTRIBUTING for image specifications.